runtime/memprof.c

```diff
@@ -183,6 +182,11 @@ enum ml_alloc_kind {
   Serialized = Val_long(2)
 };
 
+/* When we call do_callback, we suspend/resume sampling. In order to
+   to avoid unecessary calls to [caml_check_urgent_gc], we do not set
+   [caml_something_to_do] when resuming. Therefore, any call to
+   [do_callback] has to also make sure the postponed queue will be
+   handled fully at some point. */
```

Comment:
Naive question: why is it important to avoid unnecessary calls (you have a typo `unecessary` in your comment) to `caml_check_urgent_gc()` here, but not in other parts of the commit where you replaced concern-specific "do my async callbacks" by `caml_check_urgent_gc()`?

Maybe the comment should say that, in the case where `do_callback` fails with an exception, then `caml_something_to_do` is correctly set.

Reply:
In this case, an extra needless call to `caml_check_urgent_gc` will be triggered for every single call to a memprof callback. While the performance penalty of `caml_check_urgent_gc`is expected to be reasonable, I still feel like I am wasting time if such a call is issued for every single memprof callback!

Reply:
I updated the comment to make this more precise.